python_ta/transforms/z3_visitor.py

```diff
@@ -30,10 +31,10 @@ def set_function_def_z3_constraints(self, node: nodes.FunctionDef):
             if ann is None:
                 continue
             # TODO: what to do about subscripts ex. Set[int], List[Set[int]], ...
-            inferred = ann.inferred()
-            if len(inferred) > 0 and inferred[0] is not Uninferable:
-                if isinstance(inferred[0], nodes.ClassDef):
-                    types[arg.name] = inferred[0].name
+            inferred = safe_infer(ann)
+            if inferred is not Uninferable:
```

Comment:
I was initially going to suggest that this condition be combined with the next one. But actually looking at this more closely, I think `Uninferable` is not an instance of `nodes.ClassDef`, so I think it's sufficient to just check the `isinstance` part and omit the `is not Uninferable` check entirely.